### PR TITLE
HDDS-1921. TestOzoneManagerDoubleBufferWithOMResponse is flaky

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Currently `testDoubleBuffer` waits for a specific transaction count and checks `lastAppliedIndex`, which is updated last in `flushTransactions` on another thread.  I guess this may be the cause of flakiness.  By swapping the order in the test, we ensure that when the wait is over, all operations (transaction count increase, etc.) are done.

https://issues.apache.org/jira/browse/HDDS-1921

## How was this patch tested?

Ran `TestOzoneManagerDoubleBufferWithOMResponse` a few times.

### CODE REVIEW

**Comment by adoroszlai** [comment]
@bharatviswa504 please review when you have some time

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBufferWithOMResponse.java [hunk 1]
```diff
@@ -345,21 +345,23 @@ public void testDoubleBuffer(int iterations, int bucketCount)
...
+          omMetadataManager.countRowsInTable(omMetadataManager.getVolumeTable())
+      );
 
-      Assert.assertTrue(doubleBuffer.getFlushIterations() > 0);
+      Assert.assertEquals(bucketCount * iterations,
+          omMetadataManager.countRowsInTable(omMetadataManager.getBucketTable())
+      );
 
-      // Check lastAppliedIndex is updated correctly or not.
-      Assert.assertEquals((bucketCount + 1) * iterations, lastAppliedIndex);
```
Comment (bharatviswa504) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBufferWithOMResponse.java]: Why this check lastAppliedIndex is removed?

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 128 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 826 | trunk passed |
| +1 | compile | 462 | trunk passed |
| +1 | checkstyle | 106 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1178 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 226 | trunk passed |
| 0 | spotbugs | 520 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 782 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 722 | the patch passed |
| +1 | compile | 445 | the patch passed |
| +1 | javac | 445 | the patch passed |
| +1 | checkstyle | 83 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 735 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 193 | the patch passed |
| +1 | findbugs | 663 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 342 | hadoop-hdds in the patch passed. |
| -1 | unit | 2424 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 64 | The patch does not generate ASF License warnings. |
| | | 9555 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.TestStorageContainerManager |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.client.rpc.TestWatchForCommit |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.client.rpc.TestBlockOutputStream |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1238/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1238 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux f59460a8e48f 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 8cef9f8 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1238/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1238/1/testReport/ |
| Max. process+thread count | 4111 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager U: hadoop-ozone/ozone-manager |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1238/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBufferWithOMResponse.java (hunk 1 above)
Comment (xiaoyuyao) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBufferWithOMResponse.java]: The waitFor() moved above should have guarantee this condition already.

Comment (bharatviswa504) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBufferWithOMResponse.java]: Yes, you are right, I have missed it. Thanks for the pointer.

**Comment by bharatviswa504** [comment]
I will commit this to the trunk and 0.4 branch.

**Comment by adoroszlai** [comment]
Thank you @xiaoyuyao for review and @bharatviswa504 for review+commit.

**Pull request merged**